### DISCUSSION TITLE
How to edit a textbox within a group

### DISCUSSION THREAD

**Discussion by mikayelcreativefabrica** [fabricjs/fabric.js#10315] [Q&A]
Hi everyone,

I’m working on implementing functionality to change the text content of a textbox when it’s part of a group in Fabric.js.

<img width="651" alt="Screenshot 2024-12-04 at 15 14 38" src="https://github.com/user-attachments/assets/98e7011e-635b-41b5-86f6-fd056b010536">

To illustrate, I’ve shared some example code that demonstrates the approach I’m currently using with Fabric prototype mutations:
[Example on StackBlitz](https://stackblitz.com/edit/vitejs-vite-vptuzp?file=src%2Fmutationo.ts)

However, I’ve noticed that this approach is quite buggy, and I’m having issues ensuring consistent behavior.

I’m curious to know if there’s a more straightforward or efficient way to achieve this. I’d appreciate any suggestions, insights, or alternate approaches you might have!

Thanks in advance!

**Comment by asturur** [comment]
Aren't editable groups solving thise issue out of the box?

**Reply by mikayelcreativefabrica** [reply]
here [Example on StackBlitz](https://stackblitz.com/edit/vitejs-vite-rvwkqk?file=src%2Fmain.ts) I have editable group and 
drag disabled Textbox within the group, the text content editable approach works, but I need to be able to drag the group if the mouse down point intersects with the text layer and enable text content edit if the mouse up intersects with Textbox

- **Dragging the Group**: If the mouse-down event starts on the text layer and the user drags, the group should move as a whole.

- **Editing the Textbox**: If the mouse-down and mouse-up events happen on the textbox without dragging, the textbox should enter edit mode.
